worlds/zork_grand_inquisitor/world.py

```diff
@@ -0,0 +1,210 @@
+from .data.region_data import region_data
+
+from .data_funcs import (
+    item_names_to_id,
+    item_names_to_item,
+    location_names_to_id,
+    item_groups,
+    items_with_tag,
+    location_groups,
+    locations_by_region,
+    location_access_rule_for,
+    entrance_access_rule_for,
+)
+
+from .enums import (
+    ZorkGrandInquisitorEvents,
+    ZorkGrandInquisitorItems,
+    ZorkGrandInquisitorLocations,
+    ZorkGrandInquisitorRegions,
+    ZorkGrandInquisitorTags,
+)
+
+from .options import ZorkGrandInquisitorOptions
+
+
+class ZorkGrandInquisitorItem(Item):
+    game = "Zork Grand Inquisitor"
+
+
+class ZorkGrandInquisitorLocation(Location):
+    game = "Zork Grand Inquisitor"
+
+
+class ZorkGrandInquisitorWebWorld(WebWorld):
+    theme: str = "stone"
+
+    tutorials: List[Tutorial] = [
+        Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up the Zork Grand Inquisitor randomizer connected to an Archipelago Multiworld",
+            "English",
+            "setup_en.md",
+            "setup/en",
+            ["Serpent.AI"],
+        )
+    ]
+
+
+class ZorkGrandInquisitorWorld(World):
+    """
+    Zork: Grand Inquisitor is a 1997 point-and-click adventure game for PC.
+    Magic has been banned from the great Underground Empire of Zork. By edict of the Grand Inquisitor Mir Yannick, the
+    Empire has been sealed off and the practice of mystic arts declared punishable by "Totemization" (a very bad thing).
+    The only way to restore magic to the kingdom is to find three hidden artifacts: The Coconut of Quendor, The Cube of
+    Foundation, and The Skull of Yoruk.
+    """
+
+    options_dataclass = ZorkGrandInquisitorOptions
+    options: ZorkGrandInquisitorOptions
+
+    game = "Zork Grand Inquisitor"
+
+    topology_present = False
+
+    item_name_to_id = item_names_to_id()
+    location_name_to_id = location_names_to_id()
+
+    item_name_groups = item_groups()
+    location_name_groups = location_groups()
+
+    required_client_version: Tuple[int, int, int] = (0, 4, 4)
+
+    web = ZorkGrandInquisitorWebWorld()
+
+    item_name_to_item: Dict[str, ZorkGrandInquisitorItems] = item_names_to_item()
+
+    def create_regions(self) -> None:
+        deathsanity: bool = self.options.deathsanity.value == 1
+
+        region_mapping: Dict[ZorkGrandInquisitorRegions, Region] = dict()
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        for region_enum_item in region_data.keys():
+            region_mapping[region_enum_item] = Region(region_enum_item.value, self.player, self.multiworld)
+
+        region_locations_mapping: Dict[ZorkGrandInquisitorRegions, Set[ZorkGrandInquisitorLocations]]
+        region_locations_mapping = locations_by_region(include_deathsanity=deathsanity)
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        region: Region
+        for region_enum_item, region in region_mapping.items():
```

Comment:
this isn't bad but if you're ever looking for a place for speedup this is a pretty good place. the way this is being done currently is relatively slow. it's probably worth doing some of the data parsing in the location class and seeing if you can format it in such a way as to call `add_locations` with this, or alternatively pre-parse the data into a comprehension that can be more easily parsed by the AP objects.